crates/conjure_core/src/rule_engine/rewrite.rs

```diff
@@ -347,9 +351,13 @@ fn apply_all_rules<'a>(
 /// }
 ///
 fn choose_rewrite(results: &[RuleResult]) -> Option<Reduction> {
+    if results.len() > 1 {
+        bug!("There should be only 1 rule that can be applied to an expression at a time");
```

Comment:
looks good! if it actually happens we would probably also want to print the names of the rules etc.

Reply:
also: let's see if it code coverage correctly ignores this line.

Reply:
This caused a lot of integration tests to fail, does that mean there is a problem with our rules or the if statement?

Reply:
Of course! I will have it print to the console for now and once I figure out the logging will implement that there

Reply:
oh I was just meaning string concatanation to make the message you pass to bug! more descriptive. I'll check the failing tests.

Reply:
yep, we need that message to tell us a bit more :)

it looks like multiple rules were applicable all over the place. good if statement!

Reply:
I suggest we separate the two changes, as they are orthogonal. adding tracing (which is what this PR started as) is one change, adding the if statement and working on its implications should be another PR.